PracticeGit/Test.c

```diff
@@ -1,13 +1,151 @@
+      "+--------------------------------------------------+---------------+------------------+--------+\n");
+   for (int i = 0; i < testCount; i++) {
+      int size = testCases[i].size, * arrayToSort = testCases[i].array;
+      printf ("| Unsorted Array   | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      int searchValue = elementSearch[i];   // Element to search for in the current test case
+      QuickSort (arrayToSort, 0, size - 1);
+      int foundIndex = BinarySearch (arrayToSort, size, searchValue);
+      if (foundIndex != ELEMENT_NOT_FOUND) printf ("    %-11d| Found at index %-d |", searchValue, foundIndex);
+      else printf ("    %-11d|    Not Found     |", searchValue);
+      printf ("  %-8s  |\n", ArraysEqual (arrayToSort, expectedResults[i], size) ? GREEN"Pass"RESET : RED"Fail"RESET);
+      printf ("| Sorted Array     | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      printf ("\n+--------------------------------------------------+---------------+------------------+--------+\n");
+   }
+}
 
-#include "Program.h"
+void GetUserInput () {
+   int array[MAXSIZE], size;
+   char input[256];
+   printf ("Enter the number of elements (1 to %d): ", MAXSIZE);
+   fgets (input, sizeof (input), stdin);
+   size = atoi (input);     // Convert input to an integer
+   if (size < 1 || size > MAXSIZE) {
+      printf ("Invalid number of elements. Please enter a value between 1 and %d.\n", MAXSIZE);
+      return;
+   }
+   printf ("Enter %d integers:\n", size);
+   for (int i = 0; i < size; i++) {
+      while (true) {
+         fgets (input, sizeof (input), stdin);
+         char* endptr;
+         array[i] = strtol (input, &endptr, 10);
+         if (endptr != input && *endptr == '\n')  break;
+         printf ("Invalid input. Please enter an integer: ");
+      }
+   }
+   printf ("Unsorted Array: ");
+   PrintArray (array, size);
+   QuickSort (array, 0, size - 1);
+   printf ("\nSorted Array: ");
+   PrintArray (array, size);
+   int target;  // Target element to search
+   printf ("\nEnter the element to search for: ");
+   fgets (input, sizeof (input), stdin);
+   char* endptr;
+   target = strtol (input, &endptr, 10);
+   if (endptr == input || *endptr != '\n') {
+      printf ("Invalid input. Please enter a valid integer.\n");
+      return;
+   }
+   int result = BinarySearch (array, size, target);
+   printf ("Element %d found at index %d.\n", target, (result != -1) ? result : -1);
+}
 
 int main () {
-   PrintMessage ();
+   int choice;
+   do {
+      printf ("\nSelect Option:\n1. Run Test Cases\n2. Get User Input for Sorting and Searching\n"
+         "3. Exit\nEnter your choice: ");
+      char input[256];
+      fgets (input, sizeof (input), stdin);
+      choice = atoi (input);    // Get user input for menu choice
+      switch (choice) {
+      case 1:
```

Comment:
Can be written in the same line:

```suggestion
      case 1: RunQuickSortTest (); break;
      case 2: GetUserInput (); break;
      case 3: printf ("Exiting...\n"); break;
```